Storage/src/Bucket.php

```diff
@@ -779,7 +779,7 @@ public function delete(array $options = [])
      *           configuration.
      *     @type array $defaultObjectAcl Default access controls to apply to new
      *           objects when no ACL is provided.
-     *     @type array $lifecycle The bucket's lifecycle configuration.
+     *     @type array|Lifecycle $lifecycle The bucket's lifecycle configuration.
```

Comment:
Do you plan to keep allowing array?

Reply:
Yes, I didn't want to break BC.

Reply:
Does it make sense to mark the array deprecated now and remove it with the next major version?

Reply:
My gut leans towards not deprecating it - as it harms nothing to keep it around and allows users a high amount of flexibility.

Reply:
Please keep as-is in the case users would rather use an array versus the helper Lifecycle class.

Reply:
ok